### PR TITLE
ODIN_II Fix memory leaks arch/both_ram.v

### PR DESCRIPTION
**Pull request by j-b-1-7**
#### Description
Fixed memory leak in netlist_create_from_ast.cpp from not freeing in_list

#### Types of changes

- [x] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Checklist:

- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed

### CODE REVIEW

File: ODIN_II/SRC/output_blif.cpp [hunk 1]
```diff
@@ -211,6 +211,7 @@ void output_blif(const char *file_name, netlist_t *netlist)
 			}
 
 		}
+		free_nnode(node);
```
Comment (jeanlego) [inline ODIN_II/SRC/output_blif.cpp]: Freeing the node here will cause the output node to be freed, I would caution you to either free the whole netlist (nodes, net and pins) or none as we are currently doing it.

This one should be a separate PR. @ademmings

Comment (ademmings) [reply ODIN_II/SRC/output_blif.cpp]: Idk if worrying about the whole netlist is all that relevant - the pointer to this node is no longer pointing to it (is it getting replaced somewhere?) and that's the root problem. I agree with the separate PR, but I would take a different approach and try to track down where the node is actually getting lost/replaced (I don't immediately see it in this file, might have to do some digging in other files)

Comment (j-b-1-7) [reply ODIN_II/SRC/output_blif.cpp]: I removed this change from the PR. I'll try to track down where the pointer is being replaced and try to free it there
